content/en/logs/guide/restrict-access-to-log-events-with-restriction-queries.md

```diff
@@ -0,0 +1,389 @@
+```
+
+Reponse:
+
+```
+{
+	"data": [{
+		"type": "roles",
+		"id": "dcf7c550-99cb-11ea-93e6-376cebac897c",
+		"attributes": {
+			"name": "team-backend"
+		}
+	}]
+}
+```
+
+{{% /tab %}}
+{{% tab "Frontend" %}}
+
+API call:
+
+```
+curl -X POST "https://app.datadoghq.com/api/v2/logs/config/restriction_queries/b3228a0c-98fa-11ea-93e6-d30e1d2c52ee/roles" -H "Content-Type: application/json" -H "DD-API-KEY: <DATADOG_API_KEY>" -H "DD-APPLICATION-KEY: <DATADOG_APP_KEY>" -d '{"data": {"type": "roles","id": "63b970ea-99ca-11ea-93e6-e32eb84de6d6"}}’
+```
+
+Confirm that it is attached by [getting the list of roles attached to the query](https://docs.datadoghq.com/api/v2/logs-restriction-queries/#list-roles-for-a-restriction-query): 
+
+```
+curl -X GET "https://app.datadoghq.com/api/v2/logs/config/restriction_queries/b3228a0c-98fa-11ea-93e6-d30e1d2c52ee/roles" -H "Content-Type: application/json" -H "DD-API-KEY: <DATADOG_API_KEY>" -H "DD-APPLICATION-KEY: <DATADOG_APP_KEY>"
+```
+
+Reponse:
+
+```
+{
+	"data": [{
+		"type": "roles",
+		"id": "63b970ea-99ca-11ea-93e6-e32eb84de6d6",
+		"attributes": {
+			"name": "team-frontend"
+		}
+	}]
+}
+```
+
+{{% /tab %}}
+{{% tab "Generic API" %}}
+
+API call:
+
+```
+curl -X POST "https://app.datadoghq.com/api/v2/logs/config/restriction_queries/<RESTRICTION_QUERY_ID>/roles" -H "Content-Type: application/json" -H "DD-API-KEY: <DATADOG_API_KEY>" -H "DD-APPLICATION-KEY: <DATADOG_APP_KEY>" -d '{"data": {"type": "roles","id": "<ROLE_ID>"}}’
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## Attach role to the user
+
+Finally, now that your roles are configured with their permissions and restriction queries, you can give these roles to your users.
+
+### Get the user IDs
+
+The first step is to [get the list of users][10]:
+
+```
+curl -X GET "https://api.datadoghq.com/api/v2/users" -H "Content-Type: application/json" -H "DD-API-KEY: <DATADOG_API_KEY>" -H "DD-APPLICATION-KEY: <DATADOG_APP_KEY>"
+```
+
+Focus on the `data` object within the response, and extract the user IDs of the users that should belong to the `Backend` and `Frontend` role.
+
+Also, check if they already have roles and their IDs.
+
+### Attach the role to the user
+
+For each user, [assign the created backend and frontend end role][11]:
+
+```
+curl -X POST "https://api.datadoghq.com/api/v2/roles/<ROLE_ID>/users" -H "Content-Type: application/json" -H "DD-API-KEY: <DATADOG_API_KEY>" -H "DD-APPLICATION-KEY: <DATADOG_APP_KEY>" -d '{"data": {"type":"users","id":"<USER_ID>"}}'
+```
+
+### Remove default roles
+
+Users have a default Datadog role (admin, standard, read only). If this is the first time you are creating custom roles and assigning users to them, users might still have their default Datadog role, which would let them access the data.
```

Comment:
Would it make sense to move this to the beginning? This part would be important for any form of testing

Reply:
I understand your point but disagree and let me explain why.

If you start by removing default roles, this guides becomes focus for existing users. Whereas new users can be invited with a custom role.

On top of this, if you start by removing the role and do not then manage to follow all the steps (by lack of time, or anything) to create the new role and assign it to users, you create users without any roles and therefore that cannot do anything.

So by doing it at the end, you ensure that they do it after having successfully created and assigned the new roles.

Let me know if that make sense

Reply:
yup, this part convinces me.
I am just concerned that users would test incrementally as they read through the doc -- 

if we do move forward with doing nothing, can we at least add a section here https://a.cl.ly/4guykQKm
so users don't stop reading the guide preemptively and get frustrated when they havent read proper testing procedures?